logger.js

```diff
@@ -79,7 +79,7 @@ logger.dump_and_exit = function (code) {
 
 logger.log = function (level, data) {
     if (level === 'PROTOCOL') {
-        data = data.replace(/\n/g, '\\n\n');
+        data = data.replace(/\n/g, '\\n');
```

Comment:
No problems.  I don't mind a good discussion to make sure the best solution ends up in Haraka.  Feel free to disagree with me any time.

I did consider combing the logic, but decided against it because it also changes other output (like DEBUG) and I was attempting to be minimally invasive.  If other "wrapped" output (especially headers) showed up very frequently, I would definitely want to get rid of the wrapped lines.  With my configuration (with DEBUG logging on), the only output I see that has wrapped content is a couple of TLS DEBUG lines at startup and the spamassassin protocol.  I can live with the wrapped DEBUG output at startup because it doesn't happen often and honestly that wrapping is helpful.  The spamassassin output is nasty because it replays the e-mail's headers which can have unwieldy output (especially from Microsoft's crap).  Having the frequent stuff stay on one line makes it a ton easier to read the logs and diagnose problems.

Perhaps another reason not to universally combine everything on 1 line is for debugging purposes.  I have found it helpful to dump objects to the log while debugging stuff.  I would not want these object's output combined on 1 line.  However, I also wouldn't want the object's output dumped regularly in a "production" server.

To make the logs useful an application should tend to dump manageable and useful single line content and Haraka really does a good job at that already (except for the protocol piece I have brought up).

So, I'm happy with the PR I submitted.  We'll only alter the PROTOCOL to keep it on one line.  The `.replace(/\n$/, '');` line is still needed to get rid of any rogue \n chars at the end of lines because manner logger systems will add one for you and we don't need 2.  Obviously for PROTOCOL lines that replace will do nothing.

Sorry for the long speech....just spelling out my justification.